iree/test/e2e/regression/BUILD

```diff
@@ -144,7 +144,10 @@ py_binary(
         ("dylib-llvm-aot", "dylib"),
     ],
     target_cpu_features_variants = ["default"] +
-                                   (["aarch64:+dotprod"] if lhs_rhs_type == "i8" else []),
+                                   ([
+                                       "aarch64:+dotprod",
+                                       "aarch64:+i8mm",
```

Comment:
Not necessarily in this PR, but definitely soon... we should really move these tests out of `e2e/regression` and to `e2e/matmul` or something. 

https://en.wikipedia.org/wiki/Regression_testing

> is re-running [functional](https://en.wikipedia.org/wiki/Functional_testing) and [non-functional tests](https://en.wikipedia.org/wiki/Non-functional_testing) to ensure that previously developed and tested software still performs after a change.

(these are all new features being tested, not existing features that regressed at some point)

Reply:
the e2e matmul tests check that matrix multiplications don't regress. not just some particular completely new approach to implementing matrix multiplications, but matrix multiplications in general. namely, the payload being tested is a bunch of `linalg.matmul` ops, the `e2e_matmul_direct_*` tests cover whatever is the default lowering of linalg.matmul, and incidentally some additional test targets, at the moment `e2e_matmul_mmt4d_*` also cover some other non-default lowerings.  is that relevant new information ?

Reply:
I'd consider this a good rule of thumb we should aim for: if the feature lands at the same time as the tests, the tests aren't "regression tests", they're just "tests".

Reply:
but here we're talking about an existing feature, "matrix multiplication", covered by existing regression tests, and now there is this new PR that adds a new code path in its implementation, that would make it not be fully covered anymore by the existing regression tests, so this PR also adds a corresponding new variation in the regression tests. WDYT ?

Reply:
That's too broad an interpretation :) At that point, everything we're doing is a regression test as all we do is run pre-existing math and having anything but one giant flat directory is meaningless.

Regression tests are equivalent to fuzzing corpora but for humans: something worked by chance, it stopped working and thus regressed, so you check in the thing that stopped working so you don't regress again. Nothing about these matmuls should be in here unless a user reported an issue with a specific matmul that we had no coverage for and we wanted to add that specific matmul - but even then, it's always better to add an actual test in the right place instead. Large regression test suites are a smell indicating that there's not enough coverage in the proper places and something you only want to shrink over time, not grow.

Reply:
OK. IIUC, what you're saying is that in a future PR (not blocking this PR), I should move the e2e matmul tests to a new `e2e/matmul` directory as suggested by Scott. Please stop me if I'm misunderstanding.